cloud/ec2_fleet.go

```diff
@@ -19,43 +19,56 @@ import (
 	"github.com/pkg/errors"
 )
 
-type azInstanceTypeCache struct {
-	azToInstanceTypes map[string][]string
+type instanceTypeSubnetCache struct {
+	instanceTypeToSubnets map[string][]evergreen.Subnet
+	built                 bool
 }
 
-var typeCache *azInstanceTypeCache
+var typeCache *instanceTypeSubnetCache
 
 func init() {
-	typeCache = new(azInstanceTypeCache)
-	typeCache.azToInstanceTypes = make(map[string][]string)
+	typeCache = &instanceTypeSubnetCache{}
+	typeCache.instanceTypeToSubnets = make(map[string][]evergreen.Subnet)
```

Comment:
Why are you storing the whole subnet now instead of IDs? I haven't seen a place where the subnet in general is used I think

Reply:
we use the subnet ids [here](https://github.com/evergreen-ci/evergreen/pull/3810/files#diff-b6804c4bbb59072471b9ac6bac5bece9R556) and [here](https://github.com/evergreen-ci/evergreen/pull/3810/files#diff-e5707ed36e7443139f523d8951617a72R415)

Reply:
Ah what I meant was where are you using the other fields such that you can't just use map[string][]string (i.e. instance type to subnet IDs), but now we do use availability zone so 👍